File: Groove.SP/Groove.SP.Core/Entities/FreightSchedulerModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Groove.SP.Core.Entities
{
    public class FreightSchedulerModel : Entity
    {
        public long Id { set; get; }
        public string ModeOfTransport { set; get; }
        public string CarrierCode { set; get; }
        public string CarrierName { set; get; }

        public string VesselName { set; get; }
        public string Voyage { set; get; }
        public string MAWB { set; get; }
        public string FlightNumber { set; get; }

        public string LocationFromCode { get; set; }
        public string LocationFromName { set; get; }

        public string LocationToCode { get; set; }
        public string LocationToName { set; get; }

        public DateTime ETDDate { set; get; }
        public DateTime ETADate { set; get; }
        public DateTime? ATDDate { set; get; }
        public DateTime? ATADate { set; get; }

        public DateTime? CYOpenDate { set; get; }
        public DateTime? CYClosingDate { set; get; }
        
        /// <summary>
        /// If the flag is YES “Allow update from external”, IGNORE block the API update.
        /// </summary>
        public bool IsAllowExternalUpdate { get; set; }

        public virtual ICollection<ItineraryModel> Itineraries { set; get; }

        public virtual ICollection<FreightSchedulerChangeLogModel> ChangeLogs { set; get; }
    }
}

File: Groove.SP/groove-sp-web/src/app/features/freight-scheduler/models/freight-scheduler.model.ts
import { LocalDateTime } from "src/app/core/models/local-date-time.model";
import { DataType, Model, ModelBase } from "src/app/core/models/model-base.model";

@Model
export class FreightSchedulerModel extends ModelBase {
    id: number = 0;
    modeOfTransport: string = undefined;
    carrierCode: string = undefined;
    carrierName: string = undefined;
    vesselName: string = undefined;
    voyage: string = undefined;
    mawb: string = undefined;
    flightNumber: string = undefined;
    locationFromCode: string = undefined;
    locationFromName: string = undefined;
    locationToCode: string = undefined;
    locationToName: string = undefined;
    isAllowExternalUpdate: boolean = true;
    @DataType(LocalDateTime)
    etdDate: LocalDateTime = undefined;
    @DataType(LocalDateTime)
    etaDate: LocalDateTime = undefined;
    @DataType(LocalDateTime)
    atdDate: LocalDateTime = undefined;
    @DataType(LocalDateTime)
    ataDate: LocalDateTime = undefined;

    @DataType(LocalDateTime)
    cyOpenDate: LocalDateTime = undefined;
    @DataType(LocalDateTime)
    cyClosingDate: LocalDateTime = undefined;
    updatedBy: string = null;
    updatedDate: string | null = null;
    isHasLinkedItineraries:boolean = false;
}
